service/controller/v11/ebs/ebs.go

```diff
@@ -140,68 +138,48 @@ func (e *EBS) DetachVolume(ctx context.Context, volumeID string, attachment Volu
 // the Etcd volume for the master instance will be returned. If persistentVolume
 // is set then any Persistent Volumes associated with the cluster will be
 // returned.
-func (e *EBS) ListVolumes(customObject v1alpha1.AWSConfig, etcdVolume bool, persistentVolume bool) ([]Volume, error) {
-	etcdVolumeName := ""
-	volumes := []Volume{}
+func (e *EBS) ListVolumes(customObject v1alpha1.AWSConfig, filterFuncs ...func(t *ec2.Tag) bool) ([]Volume, error) {
```

Comment:
This looks heavy but is not that complicated. The interface was never really cool so I took the opportunity to improve this a bit. You can now provide filter functions. There are also some common filter functions we can reuse in certain places.